### PR TITLE
Implement get flashcards

### PR DESCRIPTION
**Pull request by filwas**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -43,11 +37,21 @@ function App() {
...
-  const handleDeleteButtonClick = (id: number) => {
-    const newCardsArray = cardsArray.filter((_, index) => index != id)
+  const handleDeleteButtonClick = (id: string) => {
+    const newCardsArray = cardsArray.filter((card) => card.id != id)
     setCardsArray(newCardsArray)
   }
 
+  
+  useEffect(() => {
+    importCards().then((importedCardsArray) => {
```
Comment (Sumick) [inline src/App.tsx]: **nitpick:** Let's name it more meaningful - like `fetchCards`, `getCards` or something similar. Import is rather reserved keyword for importing code parts across the codebase

File: src/apiService.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+export const customFetch = async <T>(
+  url: string,
+  options: RequestInit = {}
+): Promise<T> => {
```
Comment (Sumick) [inline src/apiService.ts:7]: **nitpick:** url is unused. Let's do:

```js
const BASE_URL = "https://training.nerdbord.io/api/v1/fischkapp"

export const customFetch = async <T>(
  url: string,
  options: RequestInit = {}
): Promise<T> => {
  try {
    const response = await fetch(
      BASE_URL + url,
      options
    );
```

File: src/apiService.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+interface ImportedCardObject {
+  front: string;
+  back: string;
+  _id: string;
+}
+
+export const importCards = async (): Promise<CardObject[]> => {
+  try {
+    const data = await customFetch<ImportedCardObject[]>("");
```
Comment (Sumick) [inline src/apiService.ts]: **nitpick:** Then do:

```js
const data = await customFetch<ImportedCardObject[]>("/flashcards");
```

**Review by Sumick — CHANGES_REQUESTED** [review]
Mostly looks good, however we need few improvements - please repeat this pattern into other PRs. Or finish that one, and then merge it into others

File: src/App.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ function App() {
   };
 
   useEffect(() => {
-    importCards()
+    fetchCards()
```
Comment (filwas) [inline src/App.tsx:39]: changed function name as per nitpick

File: src/App.tsx [hunk 5]
```diff
@@ -62,11 +62,11 @@ function App() {
             onCancel={handleCancelButtonClick}
           />
         )}
-        {cardsArray.map((card, index) => (
+        {[...cardsArray].reverse().map(card => (
           <Card
             faceValue={card.face}
             flipValue={card.back}
-            key={index}
+            key={card.id}
```
Comment (filwas) [inline src/App.tsx:66]: This change actually fixed a bug I had, which caused a situation where after deleting a card, the next card suddenly "inherited" the height of the deleted card. I guess this is a good example that shows how react renders and distinguishes between elements, and why keys should be unique not only to their "peers".

File: src/App.tsx [hunk 6]
```diff
@@ -31,23 +19,34 @@ function App() {
     setIsNewCardDisplayed(true);
   };
 
-  const handleSaveButtonClick = (card: CardObject) => {
-    setCardsArray([
-      { face: card.face, back: card.back },
-      ...cardsArray,
-    ]);    
+  const handleSaveButtonClick = async (card: CardObject) => {
+    await exportCard(card);
```
Comment (Sumick) [inline src/App.tsx]: **nitpick:** `Export`, is the same like `import` - it's registered keyword for doing very specific thing. We should not name `post` functions with it. Let's change it to `postNewCard` or something similar

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good! I found one naming issue + we should have only changes related to task requirements here. Now you mixed everything - we should fix it. 

TODO:

- [ ] This PR should have only task related changes.

**Review by Sumick — APPROVED** [review]
Great job!

**Pull request merged**